rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/370_profile.yml

```diff
@@ -263,7 +271,14 @@ dfs without knn vector profiling:
               keyword: "a"
 
   - match: { hits.total.value: 1 }
-  - is_false: profile.shards.0.dfs
+  - match: { profile.shards.0.dfs.statistics.type: "statistics" }
+  - match: { profile.shards.0.dfs.statistics.description: "collect term statistics" }
+  - gt: { profile.shards.0.dfs.statistics.time_in_nanos: 0 }
+  - is_true: profile.shards.0.dfs.statistics.breakdown
+  - match: { profile.shards.1.dfs.statistics.type: "statistics" }
+  - match: { profile.shards.1.dfs.statistics.description: "collect term statistics" }
+  - gt: { profile.shards.1.dfs.statistics.time_in_nanos: 0 }
```

Comment:
My understanding is where which shards the docs are on shouldn't matter as we still need to hit each shard to collect dfs information, so each shard will have a dfs profile even if no info is ultimately aggregated. (Am I missing some caching or shard skipping in this case?)

I have updated the test to change/add the following:
1. changed gt to gte for time_in_nanos
2. updated the test to add a no documents case where we still expect to get dfs profile info
3. updated the test to add a single document case where we still expect to get dfs profile info
4. updated the test to add a three document case where two share the same keyword and we still expect to get dfs profile info
5. updated the test to ensure we get no profile info if search type is `query_then_fetch`

Reply:
👍 this makes sense. For me maybe we don't need both tests (3) and (4) since having the same keyword shouldn't affect shard routing, or whether we use DFS?